dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/patch/operation/DspaceObjectMetadataRemoveOperation.java

```diff
@@ -0,0 +1,107 @@
+ * Content-Type: application/json" -d '[{ "op": "remove",
+ * "path": "/metadata/schema.identifier.qualifier(/indexOfSpecificMdToRemove)"}]'
+ * </code>
+ *
+ * @author Maria Verdonck (Atmire) on 18/11/2019
+ */
+@Component
+public class DspaceObjectMetadataRemoveOperation<R extends DSpaceObject> extends PatchOperation<R> {
+
+    @Autowired
+    DspaceObjectMetadataPatchUtils metadataPatchUtils;
+
+    @Override
+    public R perform(Context context, R resource, Operation operation) throws SQLException {
+        DSpaceObjectService dsoService = ContentServiceFactory.getInstance().getDSpaceObjectService(resource);
+        String indexInPath = metadataPatchUtils.getIndexFromPath(operation.getPath());
+        MetadataField metadataField = metadataPatchUtils.getMetadataField(context, operation);
+
+        remove(context, resource, dsoService, metadataField, indexInPath);
+        return resource;
+    }
+
+    /**
+     * Removes a metadata from the dso at a given index (or all of that type if no index was given)
+     *
+     * @param context       context patch is being performed in
+     * @param dso           dso being patched
+     * @param dsoService    service doing the patch in db
+     * @param metadataField md field being patched
+     * @param index         index at where we want to delete metadata
+     */
+    private void remove(Context context, DSpaceObject dso, DSpaceObjectService dsoService, MetadataField metadataField,
+                        String index) {
+        metadataPatchUtils.checkMetadataFieldNotNull(metadataField);
+        try {
+            if (index == null) {
+                // remove all metadata of this type
+                dsoService.clearMetadata(context, dso, metadataField.getMetadataSchema().getName(),
+                        metadataField.getElement(), metadataField.getQualifier(), Item.ANY);
+            } else {
+                // remove metadata at index
+                List<MetadataValue> metadataValues = dsoService.getMetadata(dso,
+                        metadataField.getMetadataSchema().getName(), metadataField.getElement(),
+                        metadataField.getQualifier(), Item.ANY);
+                int indexInt = Integer.parseInt(index);
+                if (indexInt >= 0 && metadataValues.size() > indexInt
+                        && metadataValues.get(indexInt) != null) {
+                    // remove that metadata
+                    dsoService.removeMetadataValues(context, dso,
+                            Arrays.asList(metadataValues.get(indexInt)));
+                } else {
+                    throw new UnprocessableEntityException("UnprocessableEntityException - There is no metadata of " +
+                            "this type at that index");
+                }
+            }
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("This index (" + index + ") is not valid nr", e);
```

Comment:
again, minor thing, but please spell out "number" instead of abbreviating to "nr".